internal/xds/bootstrap/tlscreds/bundle.go

```diff
@@ -136,3 +149,38 @@ func (c *reloadingCreds) OverrideServerName(string) error {
 func (c *reloadingCreds) ServerHandshake(net.Conn) (net.Conn, credentials.AuthInfo, error) {
 	return nil, nil, errors.New("server handshake is not supported by xDS client TLS credentials")
 }
+
+func buildSPIFFEVerifyFunc(spiffeBundleMap spiffe.BundleMap, peerVerifiedChains [][]*x509.Certificate) func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+	return func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+		// servername?
+		rawCertList := make([]*x509.Certificate, len(rawCerts))
+		for i, asn1Data := range rawCerts {
+			cert, err := x509.ParseCertificate(asn1Data)
+			if err != nil {
+				return err
+			}
+			rawCertList[i] = cert
+		}
+		leafCert := rawCertList[0]
+		roots, err := spiffe.GetRootsFromSPIFFEBundleMap(spiffeBundleMap, leafCert)
+		if err != nil {
+			return err
+		}
+
+		opts := x509.VerifyOptions{
```

Comment:
Do we need to set the `DNSName` field?

Reply:
I want to double check with you in this case what we should be using to populate DNS Name (and specifically if we are on the client side)

https://source.corp.google.com/piper///depot/google3/third_party/go/gc/src/crypto/tls/handshake_client.go;l=1135-1156;rcl=712841451

Reply:
Took a closer look: since the "normal" xDS flow (i.e. using a flat set of roots) does not set the tls.Config's ServerName field, then we should not set the DNSName field here.